task/google/resources/resource_instance_template.go

```diff
@@ -0,0 +1,226 @@
+	}
+
+	machineType := match[1]
+	if match[2] != "" {
+		acceleratorType := match[2]
+		acceleratorCount, err := strconv.Atoi(match[3])
+		if err != nil {
+			return err
+		}
+		accelerators = append(accelerators, &compute.AcceleratorConfig{
+			AcceleratorCount: int64(acceleratorCount),
+			AcceleratorType:  acceleratorType,
+		})
+	}
+
+	var isPreemptible bool
+	switch i.Attributes.Spot {
+	case 0:
+		isPreemptible = true
+	case -1:
+		isPreemptible = false
+	default:
+		return errors.New("preemptible instances don't have bidding price")
+	}
+
+	hostMaintenanceBehavior := "MIGRATE"
+	if isPreemptible || len(accelerators) > 0 {
+		hostMaintenanceBehavior = "TERMINATE"
+	}
+
+	var firewallRules []string
+	for _, rule := range i.Dependencies.FirewallRules {
+		firewallRules = append(firewallRules, rule.Identifier)
+	}
+
+	definition := &compute.InstanceTemplate{
+		Name: i.Identifier,
+		Properties: &compute.InstanceProperties{
+			MachineType: machineType,
+			Disks: []*compute.AttachedDisk{
+				{
+					Boot:       true,
+					AutoDelete: true,
+					Type:       "PERSISTENT",
+					Mode:       "READ_WRITE",
+					InitializeParams: &compute.AttachedDiskInitializeParams{
+						DiskName:    i.Identifier,
+						SourceImage: i.Dependencies.Image.Resource.SelfLink,
+						DiskSizeGb:  int64(i.Attributes.Size.Storage),
+						DiskType:    "pd-balanced",
+					},
+				},
+			},
+			NetworkInterfaces: []*compute.NetworkInterface{
+				{
+					Network: i.Dependencies.DefaultNetwork.Resource.SelfLink,
+					AccessConfigs: []*compute.AccessConfig{
+						{
+							Type:        "ONE_TO_ONE_NAT",
+							NetworkTier: "STANDARD",
+						},
+					},
+				},
+			},
+			Tags: &compute.Tags{
+				Items: firewallRules,
+			},
+			Scheduling: &compute.Scheduling{
+				OnHostMaintenance: hostMaintenanceBehavior,
+				Preemptible:       isPreemptible,
+			},
+			Labels: i.Client.Tags,
+			Metadata: &compute.Metadata{
+				Items: []*compute.MetadataItems{
+					{
+						Key:   "ssh-keys",
+						Value: &formattedPublicKey,
+					},
+					{
+						Key:   "startup-script",
+						Value: &script,
+					},
+				},
+			},
+			GuestAccelerators: accelerators,
+		},
+	}
+
+	insertOperation, err := i.Client.Services.Compute.InstanceTemplates.Insert(i.Client.Credentials.ProjectID, definition).Do()
+	if err != nil {
+		if strings.HasSuffix(err.Error(), "alreadyExists") {
+			return i.Read(ctx)
+		}
+		return err
+	}
+
+	getOperationCall := i.Client.Services.Compute.GlobalOperations.Get(i.Client.Credentials.ProjectID, insertOperation.Name)
+	_, err = waitForOperation(ctx, i.Client.Cloud.Timeouts.Create, 2*time.Second, 32*time.Second, getOperationCall.Do)
+	if err != nil {
+		return err
+	}
+
+	return i.Read(ctx)
+}
+
+func (i *InstanceTemplate) Read(ctx context.Context) error {
+	template, err := i.Client.Services.Compute.InstanceTemplates.Get(i.Client.Credentials.ProjectID, i.Identifier).Do()
+	if err != nil {
+		var e *googleapi.Error
```

Comment:
at his point I would say that this can be wrapped in the same error handler

Reply:
Not immediately actionable, but a nice suggestion. I originally wanted to write a helper function to convert a given known error into `<nil>` (some sort of error handler) but it would reduce these code blocks by a single line in the best case scenario, so, if anything, I would create a follow–up pull request.